compiler/src/iree/compiler/Dialect/Flow/TransformExtensions/FlowExtensions.cpp

```diff
@@ -693,6 +693,118 @@ transform_dialect::ClonePrecedingOpIntoDispatchRegionOp::apply(
   return DiagnosedSilenceableFailure(success());
 }
 
+// Clone a `target` op that is succeeding the given dispatch region op into the
+// dispatch region.
+//
+// All operands of the target are replaced with values defined inside of the
+// dispatch region when possible.
+//
+// If `updateUsesOutsideOfRegion` is set, all uses of the target op after the
+// dispatch region, are updated: The target op's results are returned from the
+// the dispatch region an used in those places.
+//
+// Example when `updateUsesOutsideOfRegion` is set:
+//
+// %r = flow.dispatch.region -> (tensor<?xf32>{%d0}) {
+//   %1 = "another_op"() : () -> (tensor<?xf32>)
+//   flow.return %1 : tensor<?xf32>
+// }
+// %0 = "some_op"(%r) : (tensor<?xf32>) -> (tensor<?xf32>)
+// %2 = "yet_another_use"(%0) : (tensor<?xf32>) -> (tensor<?xf32>)
+//
+// In this example, "some_op" will be cloned into the dispatch region and the
+// OpOperand of "yet_another_use" will be replaced:
+//
+// %r:2 = flow.dispatch.region -> (tensor<?xf32>{%d0}) {
+//   %1 = "another_op"() : () -> (tensor<?xf32>)
+//   %0_clone = "some_op"(%1) : (tensor<?xf32>) -> (tensor<?xf32>)
+//   flow.return %1, %0_clone : tensor<?xf32>, tensor<?xf32>
+// }
+// %2 = "yet_another_use"(%r#1) : (tensor<?xf32>) -> (tensor<?xf32>)
+static FailureOr<Flow::DispatchRegionOp> cloneSucceedingOpIntoDispatchRegion(
+    RewriterBase &rewriter, Operation *target, Flow::DispatchRegionOp regionOp,
+    bool updateUsesOutsideOfRegion) {
+  assert(regionOp->isBeforeInBlock(target) &&
```

Comment:
you dont need this assert? This is adding a huge amount of overhead to debug builds. Isnt the requirement here that the result of `regionOp` is used in `target` op? That is a much simpler check

Reply:
Well, I personally do all development in debug mode. I can live with the overhead if it provides enough value.

If this op is meant to support moving an arbitrary operation into the dispatch, then how do you know if it post-dominates/dominates the region. Then in release mode you could have SSA violation errors.

Reply:
`isBeforeInBlock` only works if the `target` and `region` operation are in the same block. I would strongly suggest only moving succeeding operations that have ssa use-def chain to make this decision. If there is no use-def chain then you need to use dominance information (that I have pointed out previously, using that during transformation is a footgun). I don't see a real use case where you would have to support such generality, and would wait for having such a case to look into such generality.